src/coreComponents/mesh/mpiCommunications/CartesianPartitioner.hpp

```diff
@@ -0,0 +1,330 @@
+   *
+   * Steps:
+   * 1. Create MPI Cartesian communicator
+   * 2. Compute local subdomain bounds
+   * 3. Determine neighbor ranks from grid topology
+   * 4. Compute checkerboard coloring
+   *
+   * @param globalMin Global domain minimum (x, y, z)
+   * @param globalMax Global domain maximum (x, y, z)
+   *
+   * @post m_neighborsRank is populated
+   * @post m_color and m_numColors are set
+   * @post Local bounds (m_localMin, m_localMax) are set
+   */
+  void initializeDomain( R1Tensor const & globalMin,
+                         R1Tensor const & globalMax ) override;
+
+  /**
+   * @brief Check if coordinate is in local Cartesian subdomain
+   *
+   * Handles periodic boundaries by mapping coordinates into
+   * the periodic range before checking bounds.
+   *
+   * @param coords Spatial coordinates (x, y, z) to test
+   * @return true if coordinate is in local partition
+   *
+   * @pre initializeDomain() must have been called
+   */
+  bool isCoordInPartition( R1Tensor const & coords ) const override;
+
+  /**
+   * @brief Check if coordinate is in partition along one axis
+   *
+   * Helper for isCoordInLocalPartition().
+   * Handles periodic boundaries.
+   *
+   * @param coord Coordinate value to test
+   * @param dir Direction (0=x, 1=y, 2=z)
+   * @return true if coordinate is in range for this direction
+   */
+  bool isCoordInPartition( real64 const & coord, int const dir ) const;
+
+
+  /**
+   * @brief Process command-line partition overrides (-x, -y, -z flags)
+   *
+   * Priority:
+   * 1. Command-line flags (if provided)
+   * 2. XML partitionCounts (if specified)
+   * 3. Default: 1 × 1 × N (1D decomposition along z-axis)
+   *
+   * @param xparCL X-direction partition count (0 = not specified)
+   * @param yparCL Y-direction partition count (0 = not specified)
+   * @param zparCL Z-direction partition count (0 = not specified)
+   *
+   * @post m_partitionCounts is set
+   * @post m_numPartitions is set
+   */
+  void processCommandLineOverrides( unsigned int const xparCL,
+                                    unsigned int const yparCL,
+                                    unsigned int const zparCL ) override;
+
+  /**
+   * @brief Set partition counts in each direction
+   *
+   * @param xPartitions Number of partitions in X direction
+   * @param yPartitions Number of partitions in Y direction
+   * @param zPartitions Number of partitions in Z direction
+   *
+   * @post m_partitionCounts = {xPartitions, yPartitions, zPartitions}
+   * @post m_numPartitions = xPartitions * yPartitions * zPartitions
+   */
+  void setPartitionCounts( unsigned int const xPartitions,
+                           unsigned int const yPartitions,
+                           unsigned int const zPartitions ) override;
+
+  /**
+   * @brief Get partition counts in all directions
+   * @return Array of partition counts {nx, ny, nz}
+   */
+  array1d< int > const & getPartitionCounts() const { return m_partitionCounts; }
```

Comment:
Why this array size is not fixed?

```suggestion
  R1Tensor const & getPartitionCounts() const { return m_partitionCounts; }
```
or
```suggestion
  stackArray1D< int, 3 > const & getPartitionCounts() const { return m_partitionCounts; }
```